File: app/src/main/java/com/example/hustmatching/viewmodel/ItemEditFrag1ViewModel.java
package com.example.hustmatching.viewmodel;

import androidx.lifecycle.ViewModel;

import java.util.ArrayList;

public class ItemEditFrag1ViewModel extends ViewModel {
    private ArrayList<String> keywords;
    private ArrayList<String> info;

    public ItemEditFrag1ViewModel() {
        keywords = new ArrayList<>();
        keywords.add("品类");
        keywords.add("品牌");
        keywords.add("外形特点");

        info = new ArrayList<>();
        info.add("地点");
    }

    public void addKeyWord(String keyword) {
        keywords.add(keyword);
    }

    public ArrayList<String> getKeywords() {
        return keywords;
    }

    public void addInfo(String information) {
        info.add(information);
    }

    public ArrayList<String> getInfo() {
        return info;
    }
}

File: app/src/main/java/com/example/hustmatching/viewmodel/ItemEditFrag1ViewModel.kt
package com.example.hustmatching.viewmodel

import android.content.Context
import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.hustmatching.bean.NetPost
import com.example.hustmatching.network.Repository
import com.example.hustmatching.room.PostDatabase
import com.example.hustmatching.utils.catch
import com.example.hustmatching.utils.checkCode
import kotlinx.coroutines.launch
import java.util.*

class ItemEditFrag1ViewModel : ViewModel() {
    val keywords: ArrayList<String>
    val info: ArrayList<String>
    val sended=MutableLiveData<Boolean>()

    //在DatePicker中选择的日期 格式：yyyy-MM-dd
    var date : String

    fun addKeyWord(keyword: String) {
        keywords.add(keyword)
    }

    fun addInfo(information: String) {
        info.add(information)
    }

    fun addPostToDatabase(post: NetPost, context: Context) {
        Thread {
            PostDatabase.getDataBase(context).postDao.insertPost(post)
        }.start()
    }

    init {
        keywords = ArrayList()
        info = ArrayList()
        //info.add("地点")
        date = ""
    }

    fun sendPosts(map: Map<String,String>){
        Log.d("map","$map")
        viewModelScope.launch {
            try {
                val response = Repository.sendPostsItem(map)
                if (response.code == 200){
                    sended.postValue(true)
                } else{
                    checkCode(response)
                }
            } catch (e: Exception) {
                catch(e)
            }
        }
    }

    fun test(){
        viewModelScope.launch {
            try {
                val response = Repository.test()
                Log.d("response data","$response")
                if (response.code != 200)
                    checkCode(response)
            } catch (e: Exception) {
                catch(e)
            }
        }
    }

}